agent/exec/container/controller.go

```diff
@@ -86,33 +86,21 @@ func (r *controller) Prepare(ctx context.Context) error {
 		return err
 	}
 
-	for {
-		if err := r.checkClosed(); err != nil {
-			return err
-		}
-
-		if err := r.adapter.create(ctx); err != nil {
-			if isContainerCreateNameConflict(err) {
-				if _, err := r.adapter.inspect(ctx); err != nil {
-					return err
-				}
-
-				// container is already created. success!
-				return exec.ErrTaskPrepared
-			}
-
-			if !engineapi.IsErrImageNotFound(err) {
-				return err
-			}
+	if err := r.adapter.pullImage(ctx); err != nil {
+		return exec.MakeTemporary(err)
```

Comment:
Should this be temporary? What if it isn't a networking problem but wrong user args?